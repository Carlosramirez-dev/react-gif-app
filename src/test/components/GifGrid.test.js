import React from 'react';
import { shallow } from "enzyme/build";
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('pruebas en el <GifGrid/>', () => {
    const category = 'Hola mundo';

    test('debe de mostrar correctamente <GifGrid/>', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wraper = shallow( <GifGrid category={category} />);
        expect( wraper ).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se cargan imagenes useFetchGifs ', () => { 
        
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'un Title'
        },
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa2.jpg',
            title: 'Otro Title'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wraper = shallow( <GifGrid category={category} />);
        expect(wraper).toMatchSnapshot();
        expect(wraper.find('p').exists()).toBe(false);
        
        expect(wraper.find('GifGridItem').length).toBe( gifs.length);
    });

        
});
