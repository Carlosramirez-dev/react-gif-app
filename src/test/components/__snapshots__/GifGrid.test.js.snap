// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <GifGrid/> debe de mostrar correctamente <GifGrid/> 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeft"
  >
    Hola mundo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Loading ...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas en el <GifGrid/> debe de mostrar items cuando se cargan imagenes useFetchGifs  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeft"
  >
    Hola mundo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="un Title"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Otro Title"
      url="https://localhost/cualquier/cosa2.jpg"
    />
  </div>
</Fragment>
`;
